aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -712,8 +712,17 @@ Tensor host_softmax(const Tensor & input_, const int64_t dim_, const bool half_t
         using accscalar_t = acc_type<scalar_t, true>;
         if (!half_to_float) {
           if (dim_size <= 1024 && dim_size*sizeof(scalar_t) <= 4096) {
-            dispatch_softmax_forward<scalar_t, scalar_t, accscalar_t, is_log_softmax>(
-                output.data_ptr<scalar_t>(), input.data_ptr<scalar_t>(), dim_size, dim_size, outer_size);
+            auto output_ptr = output.data_ptr<scalar_t>();
+            auto input_ptr = input.data_ptr<scalar_t>();
+            int64_t remaining = outer_size;
+            int64_t chunk_size = (1<<31 - 1) / dim_size;
+            while(remaining > 0) {
+              dispatch_softmax_forward<scalar_t, scalar_t, accscalar_t, is_log_softmax>(
+                output_ptr, input_ptr, dim_size, dim_size, std::min<int64_t>(remaining, chunk_size));
+              input_ptr += chunk_size * dim_size;
+              output_ptr += chunk_size;
```

Comment:
why is output advanced by a different offset? input and output sizes are the same for softmax

Reply:
This is a bug. I was thinking about `logsumexp` when writing this line, which was wrong.

Reply:
fixed